ambari-server/src/main/java/org/apache/ambari/server/state/quicklinksprofile/DefaultQuickLinkVisibilityController.java

```diff
@@ -57,6 +68,39 @@ public DefaultQuickLinkVisibilityController(QuickLinksProfile profile) throws Qu
     if (filterCount == 0) {
       throw new QuickLinksProfileEvaluationException("At least one filter must be defined.");
     }
+
+    // compute url overrides
+    profile.getFilters().stream()
+      .filter(f -> f instanceof LinkNameFilter && ((LinkNameFilter)f).getLinkUrl() != null)
+      .findAny()
+      .ifPresent(f -> LOG.warn("Link url overrides only work on service and component levels. Global filter: {}", f));
+    for (Service service : profile.getServices()) {
+      urlOverrides.putAll(getUrlOverrides(service.getName(), service.getFilters()));
+
+      for (Component component : service.getComponents()) {
+        Map<Pair<String, String>, String> componentUrlOverrides = getUrlOverrides(service.getName(), component.getFilters());
+        Sets.SetView<Pair<String, String>> duplicateOverrides = Sets.difference(urlOverrides.keySet(), componentUrlOverrides.keySet());
```

Comment:
```suggestion
        Set<Pair<String, String>> duplicateOverrides = Sets.difference(urlOverrides.keySet(), componentUrlOverrides.keySet());
```

Reply:
Also it needs to be intersection, not difference